serveradmin/serverdb/query_materializer.py

```diff
@@ -261,7 +287,11 @@ def _get_join_results(self):
             for attribute, restrict in self._attributes.items():
                 if restrict is not None:
                     servers = self._get_servers_to_join(attribute)
-                    server_objs = type(self)(servers, restrict)
+                    server_objs = type(self)(
+                        servers,
+                        restrict,
+                        attribute_lookup=self._attribute_lookup,
+                    )
```

Comment:
Ahh, here attribute_lookup is used. Presumably just so we don't have to assemble it twice?

Reply:
Yes, but I will move looking at attributes to the caller modules to make this nicer.